Description: Ignore new ASFLAGS, DFLAG in dpkg-buildflags.
Author: Alastair McKinstry <mckinstry@debian.org>
Bug-Origin: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=986516
Forwarded: no
Last-Updated: 2023-12-09

--- a/site_scons/extractdebbuild.py
+++ b/site_scons/extractdebbuild.py
@@ -28,7 +28,10 @@
 
 def getdebbuildflags():
   usedflags={'CFLAGS':None, 'CPPFLAGS':'cpp_flags', 'CXXFLAGS':'cxx_extra', 'LDFLAGS':'ld_extra'}
-  ignoreflags=['FFLAGS','FCFLAGS', 'GCJFLAGS','OBJCFLAGS','OBJCXXFLAGS']
+  ignoreflags=['FFLAGS','FCFLAGS', 'GCJFLAGS','OBJCFLAGS','OBJCXXFLAGS','DFLAGS', 'ASFLAGS',
+     'ASFLAGS_FOR_BUILD', 'CFLAGS_FOR_BUILD', 'CPPFLAGS_FOR_BUILD', 'CXXFLAGS_FOR_BUILD', 'DFLAGS_FOR_BUILD',
+     'FCFLAGS_FOR_BUILD', 'FFLAGS_FOR_BUILD', 'LDFLAGS_FOR_BUILD', 'OBJCFLAGS_FOR_BUILD', 'OBJCXXFLAGS_FOR_BUILD' ]
+
   mycflags=None
   mycxxflags=None
   try:
